### PR TITLE
Created dev mode for the oracles

### PR DESCRIPTION
**Pull request by martinvol**
## Description

Added env variable so that the node can run without node configuration.

TODO:

- [x] Mock total number of addresses so report commes faster (already done)
- [x] add mock address variable to defaults
- [x] mock expire reports

## Other changes

Added dev commands and env files.

## Tested

-

## Related issues

- 

## Backwards compatibility

All existing commands should remain the same.

### CODE REVIEW

File: src/app.ts [hunk 1]
```diff
@@ -19,6 +19,9 @@ import {
   WalletType,
 } from './utils'
 
+// const DEVMODE = true
+const MOCKADDRESS = '0x243860e8216B4F6eC2478Ebd613F6F4bDE0704DE'
```
Comment (martinvol) [inline src/app.ts:23]: will move this to env file

File: src/app.ts [hunk 2]
```diff
@@ -180,58 +182,65 @@ export class OracleApplication {
...
+          break
+        case WalletType.PRIVATE_KEY:
+          kit = newKit(httpRpcProviderUrl)
+          if (!this.config.devMode){
+            const privateKey = this.getPrivateKeyFromPath(privateKeyPath!)
+            kit.addAccount(privateKey)
+            this.config.address = privateKeyToAddress(privateKey)
+          } else{
+            this.logger.info(`DEVMODE enabled, used mock address ${this.config.mockAccount}`)
+            this.config.address = this.config.mockAccount
```
[the remark this answers is not in the document]
Comment (martinvol) [reply src/app.ts:237]: It has a default in the settings

Comment (pedro-clabs) [inline src/app.ts:237]: What's happening in this huge diff block?

File: src/reporters/base.ts [hunk 3]
```diff
@@ -418,11 +438,16 @@ export abstract class BaseReporter {
       )
     }
 
+    this.logger.info(`Starting oracle with index #${oracleIndex} ${indexOverrided? '(mocked)' : ''}`) 
+
     this._oracleIndex = oracleIndex
+    // TODO add mock here
```
Comment (pedro-clabs) [inline src/reporters/base.ts:444]: Which mock?

File: src/reporters/base.ts [hunk 4]
```diff
@@ -418,11 +438,16 @@ export abstract class BaseReporter {
...
     this._totalOracleCount =
       this.config.overrideTotalOracleCount !== undefined
         ? this.config.overrideTotalOracleCount
         : oracleWhitelist.length
+
+        this.logger.info(`Found a total of #${oracleIndex}`)
```
Comment (pedro-clabs) [inline src/reporters/base.ts:450]: `totalOracleCount`?

File: package.json [hunk 5]
```diff
@@ -10,6 +10,7 @@
     "build": "tsc -b .",
     "clean": "tsc -b . --clean",
     "start": "node lib/index.js",
+    "dev": "yarn build && . .env.dev && yarn start",
```
[the remark this answers is not in the document]
Comment (martinvol) [reply package.json:13]: It's actually quite painful to run the thing and then forget that you forgot to run `yarn build` and your changes didn't really apply.

File: src/app.ts (hunk 2 above)
Comment (martinvol) [reply src/app.ts:237]: Looks like a GitHub bug, oddly enough

File: src/data_aggregator.ts [hunk 6]
```diff
@@ -112,6 +112,8 @@ export interface DataAggregatorConfig {
    * The minimum aggregate volume across all exchanges to report
    */
   minAggregatedVolume: BigNumber
+
+  devMode: boolean
```
Comment (pedro-clabs) [inline src/data_aggregator.ts:116]: Please consider adding a comment, for consistency with the other config parameters.

**Review by pedro-clabs — APPROVED** [review]
LGTM. Thanks!

**Pull request merged**